src/app/client/src/app/modules/shared/components/on-demand-reports/on-demand-reports.component.ts

```diff
@@ -58,6 +58,9 @@ export class OnDemandReportsComponent implements OnInit {
       });
     }
   }
+  ngOnChanges(){
+    this.loadReports();
```

Comment:
Why are loading always.. It should load only when the users is in DetailsReport tab